include/deal.II/base/subscriptor.h

```diff
@@ -92,6 +92,15 @@ class Subscriptor
    */
   Subscriptor &operator = (const Subscriptor &);
 
+#ifdef DEAL_II_WITH_CXX11
+  /**
+   * Move assignment operator.
+   *
+   * Asserts that the counter for the moved object is zero.
+   */
+  Subscriptor &operator = (Subscriptor &&);
```

Comment:
This is a bit messy but I don't see any good alternative to handle this. Given the fact that ```operator = (const Subscriptor &)``` already does crazy things, I'd say it's fine.